diagnostics/reporter.rs

```diff
@@ -0,0 +1,154 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    fs,
+    hash::{DefaultHasher, Hash, Hasher},
+    io::{self, Write},
+    path::{Path, PathBuf},
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    thread,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use chrono::{DateTime, Local, Timelike};
+use concurrency::IntervalRunner;
+use logger::{debug, trace};
+use reqwest::{
```

Comment:
A use the [reqwest](https://crates.io/crates/reqwest) crate which seems to be one of the most popular and mature crates for creating REST clients.